refactor(storage-proofs): fix clippy warnings

tcharding opened the pull request:
Fix all clippy warnings produced with the following invocation:
```
cargo clippy --all-targets --all-features -- -D warnings
```

None of the fixes are particularly noteworthy except for:

- Usage of combinators instead of array index, arguably makes the test code harder to read: https://github.com/filecoin-project/rust-fil-proofs/pull/1147/commits/3592d8c78f70040c765ddaeb923e811dbe2c24c8
- Add test to criterion_group (see discussion below): https://github.com/filecoin-project/rust-fil-proofs/pull/1147/commits/889f4ad72e9d69f313ccf04b8fb2635e85a05fcf
- Allow `unit_arg` lint for usage of `black_box`: https://github.com/filecoin-project/rust-fil-proofs/pull/1147/commits/5d67be339a34ae21eb9a4d4cc5b2d475762d8465


Closes: #1123

vmx commented on storage-proofs/core/benches/drgraph.rs:
```diff
@@ -15,7 +15,8 @@ fn drgraph(c: &mut Criterion) {
                 b.iter(|| {
                     let mut parents = vec![0; 6];
-                    black_box(graph.parents(2, &mut parents).unwrap());
+                    // parents() never fails, call is_ok() to quieten clippy.
```
It's out of scope for this PR, but wouldn't it make sense create a new PR which changes `parents()` to not return a `Result`, but just nothing?

tcharding replied:
I initially thought that (even had that written in the commit log at one stage), also during the remove unwrap work this came up. `parents` is a trait method, I didn't dig into it but assumed the result was used by other implementers of the the trait, otherwise why would it be there in the first place? I can definitely look into this further now you mentioned it though.

vmx commented on storage-proofs/core/benches/pedersen.rs:
```diff
@@ -157,60 +155,6 @@ fn pedersen_circuit_benchmark(c: &mut Criterion) {
     );
 }
 
-fn pedersen_md_circuit_benchmark(c: &mut Criterion) {
```
Are you sure you want to remove this one? I'd rather add this to the `criterion_group!()` call below. You could check with the original author, if it was a mistake not adding it, or if it should be removed.

tcharding replied:
Good point, @laser `git blame` highlights you as the developer who added the macro call `criterion_group`. Do you remember if there was a reason why `pedersen_md_circuit_benchmark()` was not added to it please?

vmx commented on storage-proofs/core/src/crypto/pedersen.rs:
```diff
@@ -405,8 +405,8 @@ mod tests {
             let mut hasher = Hasher::new(&x[0]).unwrap();
-            for k in 1..5 {
-                hasher.update(&x[k]).unwrap();
+            for val in x.iter().take(5).skip(1) {
```
Just a personal nitpick (no need to change it). I would write it as `x.iter().skip(1).take(5)`. I needed a while to parse this one, I read the current version as "take the first five elements, ohhh wait, skip the first one".

tcharding replied:
Agreed.  Highlighting that you are correct is the fact that the `5` is misleading because you translated it to `x.iter().skip(1).take(5)` which iis different, we actually only want 4 items. I'll use `x.iter().skip(1).take(4)` for this and I'll use this order for this sort of combinator in future, thanks for the lesson.

tcharding commented:
Force pushed the combinator change, will wait on further comment before fixing the unused function change. Please do not merge (do you guys have a process for this, would you prefer me to change this to a draft PR to sort it out or is the discussion thread enough?) Thanks.

tcharding commented:
While we are on the topic of process, do you guys have a preference for force pushing / not force pushing? Do you follow the 'don't change history' philosophy or the 'maintain clean discrete patch sets' philosophy?

tcharding commented:
Oh, and thanks for the review @vmx, appreciate it.

laser commented:
Hi Tobin,

Nope - I have no idea why it wasn’t added. Probably an accidental omission / oversight. 

Erin

On Wed, Jun 3, 2020, at 4:23 PM, Tobin C. Harding wrote:
> 
> 
> *@tcharding* commented on this pull request.
> 
> In storage-proofs/core/benches/pedersen.rs 
> <https://github.com/filecoin-project/rust-fil-proofs/pull/1147#discussion_r434909845>:
> 
> > @@ -157,60 +155,6 @@ fn pedersen_circuit_benchmark(c: &mut Criterion) {
>      );
>  }
>  
> -fn pedersen_md_circuit_benchmark(c: &mut Criterion) {
> Good point, @laser <https://github.com/laser> `git blame` highlights 
> you as the developer who added the macro call `criterion_group`. Do you 
> remember if there was a reason why `pedersen_md_circuit_benchmark()` 
> was not added to it please?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/filecoin-project/rust-fil-proofs/pull/1147#discussion_r434909845>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAGX6G2LKG7GK7ODX5GSZHDRU3LNRANCNFSM4NRQ3HAA>.
>

vmx commented:
> Do you follow the 'don't change history' philosophy or the 'maintain clean discrete patch sets' philosophy?

I can't speak for the whole team. But I personally prefer clean discrete patch sets. Though it also has it's limits. If you e.g. do a rebase and a change and force push, it might make it hard for reviewers what actually changes. What I do in such cases is that I just push a commit with review comments addressed and once everything is reviewed a forced push with a clean history.

Havind said that, we also happily squash commits, so feel free to just keep adding things and we'll do a squash commit at the end.

tcharding commented:
> Hi Tobin, Nope - I have no idea why it wasn’t added. Probably an accidental omission / oversight. Erin

Cool, thanks.

tcharding commented:
> > Do you follow the 'don't change history' philosophy or the 'maintain clean discrete patch sets' philosophy?
> 
> I can't speak for the whole team. But I personally prefer clean discrete patch sets. Though it also has it's limits. If you e.g. do a rebase and a change and force push, it might make it hard for reviewers what actually changes. What I do in such cases is that I just push a commit with review comments addressed and once everything is reviewed a forced push with a clean history.
> 
> Havind said that, we also happily squash commits, so feel free to just keep adding things and we'll do a squash commit at the end.

Awesome, thanks.

tcharding commented:
I have updated the patch order, and amended the `criterion_group` patch based on responses above. Force pushed the whole thing and updated the initial PR comment to reflect this.

Thanks.

tcharding commented:
Thanks for the explanation @vmx, no worries at all. No rush, when you are ready just holla at me and I can rebase. I'm going to keep chipping away at little issues as I can, if there is anyway I can make your lives (as reviewers) easier just let me know.

tcharding commented:
lol you beat me to it, was just working on it now :)
